### DISCUSSION TITLE
How to add tailwind 4 alpha to sveltekit 5?

### DISCUSSION THREAD

**Discussion by traderift** [tailwindlabs/tailwindcss#13417] [Help]
This didn't work inside of vite.config.js, how can we add it?

```
import { sveltekit } from '@sveltejs/kit/vite';
import { defineConfig } from 'vite';
import tailwindcss from '@tailwindcss/vite';
export default defineConfig({
	plugins: [sveltekit(), tailwindcss()]
});
```

**Comment by Septimus** [comment]
I believe you need to use the PostCSS plugin, not the vite plugin.

**Reply by Septimus** [reply]
Ignore my previous comment. I was able to get the Tailwind 4 vite plugin working. Are you able to share a repo example for us to review and help you out?

**Reply by traderift** [reply]
Hey I added on the postcss, and it worked. However would appreciate if you could show how you've added it to the vite <3

**Comment by ghosh** [comment]
I got it working by setting vite's css transformer to lightning css. Try this:-

```ts
export default defineConfig({
    plugins: [sveltekit(), tailwindcss()],
    css: {
        transformer: 'lightningcss'
    }
});
```

**Comment by radiorambo** [comment]
Create a `./src/app.css` file  with `@import "tailwindcss";`
Then create a `./src/routes/+layout.svelte` file and import the newly-created `app.css` file with. Same as you did in tailwind v3 setup

```
<script>
	import '../app.css';
</script>
<slot />

**Comment by sjmueller** [comment]
Hi can someone share a complete (and idiomatic) example of adding tailwind 4 to a svelte 5 + sveltekit project? The comment thread so far is fragmented, and some parts misleading.

**Reply by markusstrasser** [reply]
+1

**Reply by kkirkfield** [reply]
I did not need to use lightningcss. Here are the steps I used to get this working in my projects.

1. Install Tailwind CSS v4 and the Vite plugin.

    ```sh
    npm install -D tailwindcss@next @tailwindcss/vite@next
    ```

2. Add the Tailwind CSS v4 Vite plugin to the end of vite.config.ts,

    <details>
      <summary>vite.config.ts</summary>
    
      ```ts
      import { sveltekit } from '@sveltejs/kit/vite';
      import tailwindcss from '@tailwindcss/vite';
      import { defineConfig } from 'vite';
      
      export default defineConfig({
	      plugins: [sveltekit(), tailwindcss()]
      });
      ```
    </details>

3. Add an app.css file and import Tailwind CSS. See the blog post for more ways to import layers and customize: https://tailwindcss.com/blog/tailwindcss-v4-alpha

    <details>
      <summary>src/app.css</summary>
    
      ```css
      @import 'tailwindcss';
      ```
    </details>

4. Import app.css from +layout.svelte. This example uses the new Svelte 5 runes. If you are using Svelte 4 you can use slots instead.

    <details>
      <summary>src/routes/+layout.svelte</summary>
    
      ```svelte
      <script lang="ts">
          import '../app.css'
      
          const { children } = $props();
      </script>
      
      {@render children()}
      ```
    </details>

5. (Optional) Update VS Code settings to enable Tailwind CSS v4 intellisense. Also make sure you are using the pre-release version of the VS Code extension for intellisense to work: https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss

    <details>
      <summary>.vscode/settings.json</summary>
    
      ```json
    {
	    "editor.quickSuggestions": {
		    "strings": "on"
	    },
	    "files.associations": {
		    "*.css": "tailwindcss"
	    }
    }
      ```
    </details>

**Reply by sjmueller** [reply]
tysm @kkirkfield 🙏 clear as day, and just what the svelte community needed to be inspired and give it a try

**Reply by lucas-subli** [reply]
This works overall. But the @apply syntax seems to be broken ([Github issue](https://github.com/tailwindlabs/tailwindcss/issues/13305)). 

Using a framewordk in RC stage with an ALPHA build of tailwind is prone to errors, who would have guessed...

**Reply by lucas-subli** [reply]
There is an ongoing discussion on @apply at https://github.com/tailwindlabs/tailwindcss/discussions/15205

**Comment by vnadhan** [comment]
Hello! I would like your help in fixing an issue on integrating tailwind v4.1.11 with Svelte5. When I run `npm run build` for a production build, the build freezes on `transforming...` as you see below and this is because vite is unable to figure out the `tailwindcss` import

```text
 => => # vite v6.3.5 building SSR bundle for production...                                                                                                                                                                                     
 => => # transforming...                                                                                                                                                                                                                       
 => => # 2025-07-23T09:24:13.913Z vite:resolve 0.17ms tailwindcss -> null  
```

this is my vite.config.ts

```js
import { sveltekit } from '@sveltejs/kit/vite';
import { defineConfig } from 'vite';
import tailwindcss from '@tailwindcss/vite';

export default defineConfig({
	plugins: [tailwindcss(), sveltekit()],
	ssr: {
		noExternal: [
			'@smui/**',
		]
	}
});
```

and there is a file called `src/aseets/global.css` which has the import `@import 'tailwindcss';` and this file is imported in `layout.svelte` as `import tailwindCssUrl from '$lib/assets/global.css?url';`
